In [1]:
import numpy as np
import tensorflow as tf

With out GPU Support



In [3]:
import time

In [4]:
tf.test.gpu_device_name()

''

In [5]:
shape=5000
np_array = np.random.uniform(size = [shape,shape])
start = time.time()

for i in range(10):
  np.matmul(np_array, np_array)

end = time.time()
time_period = end-start

print(f'{time_period} seconds')

79.02612090110779 seconds


In [7]:
shape=5000
tf_array = tf.random.uniform([shape,shape])
start = time.time()

for i in range(10):
  tf.matmul(tf_array, tf_array)

end = time.time()
time_period = end-start

print(f'{time_period} seconds')

41.416791915893555 seconds


with GPU Support

In [1]:
import numpy as np
import tensorflow as tf

In [2]:
import time

In [3]:
tf.test.gpu_device_name()

'/device:GPU:0'

In [4]:
shape=5000
np_array = np.random.uniform(size = [shape,shape])
start = time.time()

for i in range(10):
  np.matmul(np_array, np_array)

end = time.time()
time_period = end-start

print(f'{time_period} seconds')

85.03747892379761 seconds


In [5]:
shape=5000
tf_array = tf.random.uniform([shape,shape])
start = time.time()

for i in range(10):
  tf.matmul(tf_array, tf_array)

end = time.time()
time_period = end-start

print(f'{time_period} seconds')

0.20374107360839844 seconds


With out GPU
numpy = 79.04
tensor = 41.41

with GPU
numpy = 85.03
tensor = 0.20 Seconds

Tensor is too much matchn and speed in GPU
